### PR TITLE
Add a document with known-issues

### LINKED ISSUE #2972: Folder Redirection Windows
### Description

I'm using a Win7 machine at work and it is configured with folder redirection.

https://technet.microsoft.com/en-us/library/cc732275(v=ws.11).aspx
https://en.wikipedia.org/wiki/Folder_redirection

This configuration forces Github Desktop to install at `\\NETWORK\Redirected\USERNAME\Application Data\GitHub Desktop`. The application opens, but much of the functionality is broken because it has trouble finding and using the data in the folder.

I had the same issue with Atom and I created this issue https://github.com/atom/atom/issues/15663. We were able to workaround by creating an 'electronUserData' folder at `C:\Users\USERNAME\.atom`.

Is it possible to do something similar with Github Desktop to restore functionality?

Thanks for your help.

### Version

**GitHub Desktop version:** [1.0.3]

**OS version:** [Microsoft Windows 6.1.7601]

### Logs
[2017-09-21.desktop.production.log](https://github.com/desktop/desktop/files/1360553/2017-09-21.desktop.production.log)
[2017-09-22.desktop.production.log](https://github.com/desktop/desktop/files/1360554/2017-09-22.desktop.production.log)
[2017-10-05.desktop.production.log](https://github.com/desktop/desktop/files/1360558/2017-10-05.desktop.production.log)

### LINKED ISSUE #4039: Consider using installed Git binaries instead of bundled git/msys-2.0.dll
### Description

Currently GitHub desktop uses bundled git mingw binaries (especially `msys-2.0.dll`). It causes some issues if there is a different version of  `msys-2.0.dll` on the system already installed (likely the case if user has installed Git for Windows).  This causes issues like #3096 . Since multiple versions of the said dll causes  cygheap problems, Avoid using bundled git/mingw when a Git installation is found on the system. 

### Version

**GitHub Desktop version:** 1.0.13

**OS version:** Microsoft Windows [Version 10.0.16299.248]

### Steps to Reproduce

1. Install 32 bit Git
2. Install 64 Bit GitHub Desktop app
3. Clone a repository /commit to a repository & you will get cygheap errors.

The issue is not specific to the 32 bit / 64 bit Github desktop setup case, it happens whenever there are multiple versions of msys-2.0.dll.

**Expected behavior:** Use installed git binaries /dlls instead of bundled git whenever possible.

**Actual behavior:** the app currently used bundled binaries

**Reproduces how often:** Always with different version of `msys-2.0.dll`. If all the dlls are of same version and arch, then everything works fine.

### PR DESCRIPTION
**Pull request by nerdneha**
**[Rendered Markdown](https://github.com/desktop/desktop/blob/known-issues-doc/docs/known-issues.md)**

Purpose: a way for us to track "acknowledged" issues without clogging the issue tracker. 
How to use: If you want to close a known issue, add it to this doc with some context

These issues may or may not have workarounds. 

- [x] have README point to it
- [x] add some context (part 2 of doc) around what to do if....

cc @steveward, re: starting to document known issues and having a place to point people to

Closes #2972 
Closes #4039

### CODE REVIEW

**Comment by shiftkey** [comment]
@nerdneha just took a shot at structuring the doc and answering the second item on your checklist - feel free to adjust or adapt it to suit

File: docs/known-issues.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+**Workaround:** 
+
+ - Remove `%APPDATA%\GitHub Desktop\window-state.json` 
+ - Restart Desktop
+
+### Using a repository configured with Folder Redirection - [#2972](https://github.com/desktop/desktop/issues/2972)
+
+[Folder Redirection](https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/cc753996(v%3dws.11)) is an feature of Windows for administrators to ensure files and folders are managed on a network server, instea.
+
+**Not supported** as Git is not able to resolve the working directory correctly:
```
Comment (nerdneha) [inline docs/known-issues.md]: not supported initially made me still feel like we might have a workaround. it's clear to me that we don't have one from the desktop side but maybe we can use a different phrase for this to be clearer?

some options
* no workaround
* unable to fix
* fix dependent on _____ (ex: git)

**Comment by nerdneha** [comment]
fyi i added a "close" for the two added issues in this PR description; feel free to edit+take out if we want to keep any of the 3 issues open @shiftkey

**Comment by shiftkey** [comment]
@nerdneha I dropped the #3263 mention because it's already closed, and I think we can do better than the workaround in #2107 and I have access to a spare monitor at the moment so I'm going to see if I can address that separate to this.

**Comment by shiftkey** [comment]
:apple:

**Comment by shiftkey** [comment]
Added a couple of Windows-specific issues, but this can be reviewed and merged at any time as it's not code-related...

**Review by iAmWillShepherd — APPROVED** [review]

**Review by macotouenaca — COMMENTED** [review]
稼ぐ

**Pull request merged**